.github/workflows/deploy-to-eks.yaml

```diff
@@ -0,0 +1,59 @@
+# USAGE
+# -----
+# # NOTE: Can be used as part of other workflows, such as ci-ecr.yaml
+# on:
+#   workflow_dispatch:
+#     branches:
+#       - main
+#   push:
+#     branches:
+#       - main
+#     paths-ignore:
+#       - "Jenkinsfile"
+#       - ".git**"
+#
+# jobs:
+#   deploy-to-eks:
+#     uses: alphagov/govuk-infrastructure/.github/workflows/deploy-to-eks.yaml@main
+#     secrets:
+#       CI_GITHUB_API_TOKEN: ${{ secrets.CI_GITHUB_API_TOKEN }}
+
+# REUSABLE WORKFLOW
+# -----------------
+
+name: Deploy to EKS
+
+on:
+  workflow_call:
+    secrets:
+      CI_GITHUB_API_TOKEN: ${{ secrets.CI_GITHUB_API_TOKEN }}
+
+jobs:
+  deploy:
+    name: Update image tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        env:
+          GIT_USER_EMAIL: govuk-ci@digital.cabinet-office.gov.uk
+          GIT_USER_NAME: govuk-ci
+          GITHUB_TOKEN: ${{ secrets.GOVUK_CI_GITHUB_API_TOKEN }}
+          GOVUK_ENV: integration
+          IMAGE_TAG: ${{ github.sha }}
+          REPOSITORY: ${{ github.event.repository.name }}
+        run: |
+          set -eu
+          git config --global user.email "${GIT_USER_EMAIL}"
+          git config --global user.name "${GIT_USER_NAME}"
+          echo "Modifying Helm Charts repo..."
+          repo="https://${GIT_USER_NAME}:${GITHUB_TOKEN}@github.com/alphagov/govuk-helm-charts.git"
+          branch="main"
+          file="charts/argocd-apps/image-tags/${GOVUK_ENV}/${REPOSITORY}"
+          git clone --single-branch --depth 1 --branch main $repo
+          cd "govuk-helm-charts"
+          echo $IMAGE_TAG > $file
```

Comment:
optional: some of these e.g. `$file` should strictly have quotes (like `"${file}"`) for robustness but we know we're not going to be dealing with spaces here, and we're not running shellcheck on this so 🤷